scene/2d/skeleton_2d.cpp

```diff
@@ -429,29 +429,22 @@ PackedStringArray Bone2D::get_configuration_warnings() const {
 }
 
 void Bone2D::calculate_length_and_rotation() {
-	// if there is at least a single child Bone2D node, we can calculate
+	// If there is at least a single child Bone2D node, we can calculate
 	// the length and direction. We will always just use the first Bone2D for this.
-	bool calculated = false;
 	int child_count = get_child_count();
-	if (child_count > 0) {
-		for (int i = 0; i < child_count; i++) {
-			Bone2D *child = Object::cast_to<Bone2D>(get_child(i));
-			if (child) {
-				Vector2 child_local_pos = to_local(child->get_global_position());
-				length = child_local_pos.length();
-				bone_angle = child_local_pos.normalized().angle();
-				calculated = true;
-				break;
-			}
+
+	for (int i = 0; i < child_count; i++) {
+		Bone2D *child = Object::cast_to<Bone2D>(get_child(i));
+		if (child) {
+			Vector2 child_local_pos = to_local(child->get_global_position());
+			length = child_local_pos.length();
+			bone_angle = child_local_pos.normalized().angle();
```

Comment:
It's probably better to do:  

```cpp
length = (child->get_global_position() - get_global_position()).length();
bone_angle = to_local(child->get_global_position()).angle();
```

This will give us length scaled and probably help with scaling problems from others issues.

Reply:
Bone shapes would probably need to convert to absolute, because it uses `RenderingServer::get_singleton()->canvas_item_add_polygon()` to draw it and seems to take scale sign in count somehow.

Reply:
True, no intention of doing changes about this here, just registering thoughts (would have to review many `Skeleton2DModification` to actually make this change).

> If changed, it should be properly justified why the given change is needed.
> > Bone shapes would probably need to convert to absolute, because it uses `RenderingServer::get_singleton()->canvas_item_add_polygon()` to draw it and seems to take scale sign in count somehow.

I need to check if this works as I think before justifying anything.